_posts/2023-05-22-grpc-subsystem.adoc

```diff
@@ -0,0 +1,273 @@
+---
+layout: post
+title:  "grpc Subsystem in WildFly"
+date:   2023-05-22
+tags:   grpc
+author: rsigal, hpehl
+description: Introducing grpc subsystem in WildFly
+---
+
+We are pleased to announce the first release of support for
+https://grpc.io/[*gRPC*] services in WildFly. gRPC, a Google project,
+is, as its name suggests, a Remote Procedure Call framework. In some
+ways, it is a competitor to
+https://jakarta.ee/specifications/restful-ws/[*Jakarta RESTFul Web
+Services*], but it has different semantics and a different performance
+profile.
+
+gRPC rests on another Google project, https://protobuf.dev/[*protobuf*],
+which is, according to the website, "a language-neutral,
+platform-neutral extensible mechanism for serializing structured data."
+It consists of 1) a programming language independent data definition
+language and 2) a compressed wire protocol for transmitting data.
+Moreover, protobuf data is transported over HTTP/2, and taken together,
+gRPC is well positioned for speedy transmission. On the other hand,
+the most common data format in the Jakarta REST world is the more
+readable but more verbose JSON, and there is evidence
+(https://medium.com/@EmperorRXF/evaluating-performance-of-rest-vs-grpc-1b8bdf0b22da[*Evaluating
+Performance of REST vs. gRPC*]) that gRPC transmission can be
+considerably faster than Jakarta REST transmission. Of course, your
+mileage may vary.
+
+Given that this is a WildFly blog, and given that RESTEasy, as an
+implementation of Jakarta RESTful Web Services,
+is a foundational technology in WildFly,
+we will assume that the reader is familiar with Jakarta REST but not
+necessarily familiar with gRPC, and so we’ll take a minute to introduce
+the latter.
+
+== What is gRPC?
+
+gRPC is a modern iteration in a long history of RPC frameworks, a
+notable example of which is CORBA. In particular, as one smart guy
+saidfootnote:[Stuart Douglas, email], "gRPC is to CORBA what REST is to
+SOAP", which should be a relief to anyone proposing to learn gRPC.
+
+A gRPC application begins with a language neutral description of data
+types and procedure calls. Consider, for example, the following, which
+comes from the _helloworld_ example in the wildfly-grpc-feature-pack
+project (https://github.com/wildfly-extras/wildfly-grpc-feature-pack):
+
+[source,bash]
```

Comment:
Hey, @ronsigal, the code syntax seems to be `[source,bash]` for all code samples.